sql/2024/accessibility/alt_ending_in_image_extension.sql

```diff
@@ -0,0 +1,59 @@
+#standardSQL
+# Alt text ending in an image extension
+
+# Temporary function to extract file extension statistics from the JSON payload
+CREATE TEMPORARY FUNCTION getUsedExtensions(payload STRING)
+RETURNS ARRAY<STRUCT<extension STRING, total INT64>> LANGUAGE js AS '''
+try {
+  const a11y = JSON.parse(payload);
+
+  return Object.entries(a11y.file_extension_alts.file_extensions).map(([extension, total]) => {
+    return {extension, total};
+  });
+} catch (e) {
+  return [];
+}
+''';
+
+# Main query to analyze alt text usage across sites
+SELECT
+  client,  # Client domain
+  COUNTIF(CAST(JSON_EXTRACT_SCALAR(JSON_EXTRACT_SCALAR(payload, '$._markup'), '$.images.img.alt.present') AS INT64) > 0) AS sites_with_non_empty_alt,  # Count sites with non-empty alt attributes
```

Comment:
The custom metrics (`$._XXXX`) are in their own column in the new schema so don't need to be pulled from the larger `payload` column. Note these don't have the leading underscore so that needs to be removed too.

```suggestion
  COUNTIF(CAST(JSON_EXTRACT_SCALAR(JSON_EXTRACT_SCALAR(custom_metrics, '$.markup'), '$.images.img.alt.present') AS INT64) > 0) AS sites_with_non_empty_alt,  # Count sites with non-empty alt attributes
```

Can you fix all other instances of this? In this query and elsewhere.

Also not there is currently an issue with this column for 2024-06-01 so don't run these just yet (but do update them to be ready). Follow here for updates:
https://httparchive.slack.com/archives/C016QLMT0BZ/p1724280196278609